src/constants/CODATA2018/test/CODATA2018.test.cpp

```diff
@@ -0,0 +1,96 @@
+#define CATCH_CONFIG_MAIN
+
+#include <fstream>
+#include <typeinfo>
+
+#include "catch.hpp"
+#include "Log.hpp"
+
+#include "constants.hpp"
+#include "constants/CODATA2018.hpp"
+#include "constants/testing.hpp"
+
+using namespace njoy::constants;
+using namespace dimwits;
+
+auto stringFor = hana::make_map(
+  hana::make_pair( avogadro, std::string{ "Avogadro constant" } ),
+  hana::make_pair( boltzmann, std::string{ "Boltzmann constant" } ),
+  hana::make_pair( electric, std::string{ "vacuum electric permittivity" } ),
+  hana::make_pair( electronRadius, std::string{ "classical electron radius" } ),
+  hana::make_pair( elementaryCharge, std::string{ "elementary charge" } ),
+  hana::make_pair( faraday, std::string{ "Faraday constant" } ),
+  hana::make_pair( fineStructure, std::string{ "fine-structure constant" } ),
+  hana::make_pair( conductanceQuantum, std::string{ "conductance quantum" } ),
+  hana::make_pair( magnetic, std::string{ "vacuum mag. permeability" } ),
+  hana::make_pair( magneticFluxQuantum, std::string{ "mag. flux quantum" } ),
+  hana::make_pair( molarGas, std::string{ "molar gas constant" } ),
+  hana::make_pair( newtonianGravitation, 
+                   std::string{ "Newtonian constant of gravitation" } ),
+  hana::make_pair( planck, std::string{ "Planck constant" } ),
+  hana::make_pair( hbar, std::string{ "reduced Planck constant" } ),
+  hana::make_pair( rydberg, std::string{ "Rydberg constant" } ),
+  hana::make_pair( speedOfLight, std::string{ "speed of light in vacuum" } ),
+  hana::make_pair( stefanBoltzmann, std::string{ "Stefan-Boltzmann constant" } ),
+
+  // Masses
+  hana::make_pair( electronMass, std::string{ "electron mass" } ),
+  hana::make_pair( neutronMass, std::string{ "neutron mass" } ),
+  hana::make_pair( protonMass, std::string{ "proton mass" } ),
+  hana::make_pair( deuteronMass, std::string{ "deuteron mass" } ),
+  hana::make_pair( helionMass, std::string{ "helion mass" } ),
+  hana::make_pair( tritonMass, std::string{ "triton mass" } ),
+  hana::make_pair( alphaMass, std::string{ "alpha particle mass" } )
+);
+
+template< typename MAP >
+void checkMap( MAP& map ){
+  auto referenceValues = 
+      testing::defineReferenceValues( std::ifstream("CODATA2018.txt") );
+
+  hana::for_each( hana::keys( map ), [&]( auto&& key ){
+    auto refKey = stringFor[ key ];
+
+    GIVEN( "constant: " + refKey ){
+      auto reference = referenceValues[ refKey ];
+
+      // Hiding our shame
+      auto verifyIfExists = [&](auto key) {
+        return hana::overload(
+          []( hana::true_ ){ return true; },
+          [&]( auto value ){ return reference.second == value.value; } 
+        )( hana::find( map.uncertainty, key ).value_or( hana::true_c ) );
+      };
+
+      // njoy::Log::info( "{:35s}\tref.first: {:.12G}, map.value: {:.12G}", 
+      //                  refKey, reference.first, map[ key ].value );
+
+      CHECK( fabs( 1 - (reference.first/map[ key ].value ) ) < 7E-10 );
```

Comment:
Note that in [CODATA2014.test.cpp](https://github.com/njoy/constants/blob/feature/CODATA2018/src/constants/CODATA2014/test/CODATA2014.test.cpp#L63) the comparison value is `5E-10`. I had to increase this one to `7E-10` to get hbar or `ReducedPlanck` to pass for CODATA2018.

In the [reference file](https://github.com/njoy/constants/blob/feature/CODATA2018/src/constants/CODATA2014/test/CODATA2014.test.cpp#L63) downloaded from NIST the "reduced Planck constant" is only has 10 significant digits (followed by `...`). The ninth decimal place is `7` in the official version, but our derived value rounds up to an `8` in that spot. It seems to me that they didn't round quite correctly, but that's what we have to live with.

Reply:
Probably a really basic question, but what does the "(exact)" Uncertainty value mean in the reference file if the value isn't, well, exact? I'd have thought that an "exact" value would be, say, a value like the speed of light in meters/second, which does indeed have an exact value. Is the unspecified "..." enough to characterize it as exact?

Reply:
Does NIST have any sort of correction process for their data, or do we genuinely need to live with it? If we do have to live with it, is there perhaps a methodology in your codes that would allow a user to request our "corrected" [we hope] value?

Reply:
`(exact)` means just what you think it does. It means that the international community has decided that the value is defined and has no uncertainty. I’m not exactly sure what the `...` means. I suspect it means there are more digits, but they are not shown. I think most of the constants with `...` are derived values  which are calculated from others.

I’m not sure what the process is for suggesting changes to the NIST values. CODATA is an international organization and we generally just accept what they have. Now, if someone finds an error in our values, they can (and are encouraged to) create an issue and let us know so that we can fix our value to match the values for CODATA.